wrappers/bedtools/bamtofastq/wrapper.py

```diff
@@ -0,0 +1,16 @@
+from snakemake.shell import shell
+
+log = snakemake.log_fmt_shell(stdout=True, stderr=True)
+
+outdir = snakemake.output.fastq1.split('/')[0]
+
+shell(
+
+    "(samtools sort -n  "
```

Comment:
A sorted bam is definitely needed before conversion to fastq. 

But It might be better if this was optional in the wrapper so it gives the pipeline developer the option of handing sorted bams. The wrapper could take in a bool asking if sorting should happen. If yes, append the string 

"samtools sort -n .... | "

to the beginning of the shell string. If not, the shell string should just be "bamtofastq ..."

Reply:
Is there an easy way for a user to know if the bam is sorted by read name? Usually bams are sorted by chromosomal coordinate, but bamtofastq [requires sorting by read name ](https://bedtools.readthedocs.io/en/latest/content/tools/bamtofastq.html), at least for paired-end reads, which will be the vast majority of cases.

Reply:
Yes, it looks like in the bam header, there's a value that should be set called SO (sort order):

@HD     VN:1.6  SO:coordinate

https://samtools.github.io/hts-specs/SAMv1.pdf
https://www.biostars.org/p/5256/

So grepping for "coordinate" on the header is one option. 

I agree that sorting beforehand would coverage a majority of cases but going in line with the Snakemake philosophy a wrapper should be agnostic to the pipeline that it is being used on. 

Wrappers should be written such that you could basically drop the tool into any pipeline and it would have enough customizability to suit that pipeline's needs. Take a look at the wrappers in the snakemake-wrappers repo -- those have extensive customization options and that's why we can just outright copy them over into crg2. Sorry for being so nitpicky.

Reply:
Smart! OK, I'll change this.

Reply:
added in option to skip sorting if bam is already qsorted